openverse_catalog/dags/provider_api_scripts/wikimedia_commons.py

```diff
@@ -213,32 +222,90 @@ def _merge_image_pages(left_page, right_page):
     return merged_page
 
 
-def _process_image_data(image_data):
+def _extract_file_type(media_info):
+    filetype = media_info.get("mimetype", "").split("/")[-1]
+    if not filetype:
+        filetype = media_info.get("url", "").split(".")[-1]
+    return None if filetype == "" else filetype
+
+
+def _process_media_data(image_data):
     foreign_id = image_data.get("pageid")
     logger.debug(f"Processing page ID: {foreign_id}")
-    image_info = _get_image_info_dict(image_data)
-    valid_mediatype = _check_mediatype(image_info)
+    media_info = _get_image_info_dict(image_data)
+    valid_mediatype = _check_mediatype(media_info)
     if not valid_mediatype:
         return None
-    license_info = _get_license_info(image_info)
+    license_info = _get_license_info(media_info)
     if license_info.url is None:
         return None
-    image_url = image_info.get("url")
-    creator, creator_url = _extract_creator_info(image_info)
-    title = _extract_title(image_info)
-
-    image_store.add_item(
-        foreign_landing_url=image_info.get("descriptionshorturl"),
-        image_url=image_url,
-        license_info=license_info,
-        foreign_identifier=foreign_id,
-        width=image_info.get("width"),
-        height=image_info.get("height"),
-        creator=creator,
-        creator_url=creator_url,
-        title=title,
-        meta_data=_create_meta_data_dict(image_data),
-    )
+    media_url = media_info.get("url")
+    creator, creator_url = _extract_creator_info(media_info)
+    title = _extract_title(media_info)
+    filesize = media_info.get("size", 0)  # in bytes
+    filetype = _extract_file_type(media_info)
+    parsed_data = {
+        "foreign_landing_url": media_info.get("descriptionshorturl"),
+        "foreign_identifier": foreign_id,
+        "license_info": license_info,
+        "creator": creator,
+        "creator_url": creator_url,
+        "title": title,
+        "filetype": filetype,
+        "filesize": filesize,
+    }
+
+    if valid_mediatype == IMAGE:
+        _add_image(parsed_data, media_url, image_data, media_info)
+    else:
+        _add_audio(parsed_data, media_url, image_data, media_info)
```

Comment:
Possible alternative:
```suggestion
    funcs = {
        IMAGE: _add_image,
        AUDIO: _add_audio,
    }
    funcs[valid_mediatype](parsed_data, media_url, image_data, media_info)
```

This makes it easy to expand later (if we end up doing more than just image/audio) and will cause a hard stop if we get something different, rather than jumping into the `else` condition and assuming audio with possibly erroneous data.